client/src/partials/exchange/modal.js

```diff
@@ -27,10 +39,24 @@ function ExchangeModalController(Rates, Currencies, Session, $uibModalInstance,
 
   function submit(invalid) {
     if (invalid) { return; }
+    vm.data.enterprise_id = vm.enterpriseCurrencyId;
```

Comment:
This line tells me that the enterprise_id === enterprise currency id.  Why?  That doesn't seem right.